data-dashboard/lib/components/InputField/InputField.tsx

```diff
@@ -0,0 +1,42 @@
+// This component can be used to wrap any input component and add a label
+// to it while vinculating them together. It should accept different input
+// components, not just TextInput, but there aren't any other input
+// components yet, please add them here if you create them.
+
+import styled from 'styled-components';
+import TextInput, { TextInputProps } from '../TextInput/TextInput';
+
+interface InputFieldProps {
+  label: string;
+  name?: string;
+  inputChild: typeof TextInput;
+  inputProps: TextInputProps;
```

Comment:
I am not a huge fan of this pattern but this line makes it possible to skip type `TextInputProps`
```suggestion
  inputProps: React.ComponentProps<typeof TextInputProps>;
```

Reply:
I will surely add this later with other changes.